test/e2e/kubectl/logs.go

```diff
@@ -209,4 +255,49 @@ var _ = SIGDescribe("Kubectl logs", func() {
 		})
 	})
 
+	ginkgo.Describe("all pod logs", func() {
+		ginkgo.Describe("the StatefulSet has 2 replicas and each pod has 2 containers", func() {
+			var sts *appsv1.StatefulSet
+			stsName := "sts" + string(uuid.NewUUID())
+			numberReplicas := int32(2)
+			ginkgo.BeforeEach(func(ctx context.Context) {
+				stsClient := c.AppsV1().StatefulSets(ns)
+				ginkgo.By("constructing the StatefulSet")
+				stsCopy := testingStatefulSet(stsName, ns, numberReplicas)
+				sts = &stsCopy
+				ginkgo.By("creating the StatefulSet")
+
+				_, err := stsClient.Create(ctx, sts, metav1.CreateOptions{})
+				if err != nil {
+					framework.Failf("Failed to create StatefulSet: %v", err)
+				}
+			})
+
+			ginkgo.AfterEach(func() {
+				e2ekubectl.RunKubectlOrDie(ns, "delete", "sts", stsName)
+			})
+
+			ginkgo.It("should get logs from all pods", func(ctx context.Context) {
+				ginkgo.By("Waiting for StatefulSet pods to be running.")
+				e2estatefulset.WaitForStatusAvailableReplicas(ctx, c, sts, int32(numberReplicas))
+
+				ginkgo.By("default container for each pod")
+				out := e2ekubectl.RunKubectlOrDie(ns, "logs", fmt.Sprintf("sts/%s", stsName), "--all-pods")
+				framework.Logf("got output %q", out)
+				gomega.Expect(out).NotTo(gomega.BeEmpty())
+				gomega.Expect(out).To(gomega.ContainSubstring("container-2"))
+				gomega.Expect(out).NotTo(gomega.ContainSubstring("container-1"))
+
+				ginkgo.By("all containers for each pod")
+				out = e2ekubectl.RunKubectlOrDie(ns, "logs", fmt.Sprintf("sts/%s", stsName), "--all-pods", "--all-containers")
+				framework.Logf("got output %q", out)
+				gomega.Expect(out).NotTo(gomega.BeEmpty())
+				gomega.Expect(out).To(gomega.ContainSubstring("container-2"))
+				gomega.Expect(out).To(gomega.ContainSubstring("container-1"))
```

Comment:
I think we're missing a validation here.

In the first test can we please confirm that there are 2 pods and in the second test confirm that there are 4 containers? Does that sound right?